usage.go

```diff
@@ -78,26 +78,37 @@ func (p *Parser) WriteHelp(w io.Writer) {
 	}
 
 	// write the list of options
-	if len(options) > 0 {
-		fmt.Fprint(w, "\noptions:\n")
-		const colWidth = 25
-		for _, spec := range options {
-			left := "  " + synopsis(spec, "--"+spec.long)
-			if spec.short != "" {
-				left += ", " + synopsis(spec, "-"+spec.short)
-			}
-			fmt.Fprint(w, left)
-			if spec.help != "" {
-				if len(left)+2 < colWidth {
-					fmt.Fprint(w, strings.Repeat(" ", colWidth-len(left)))
-				} else {
-					fmt.Fprint(w, "\n"+strings.Repeat(" ", colWidth))
-				}
-				fmt.Fprint(w, spec.help)
-			}
-			fmt.Fprint(w, "\n")
+	fmt.Fprint(w, "\noptions:\n")
+	for _, spec := range options {
+		printOption(w, spec)
+	}
+
+	// write the list of built in options
+	dest := reflect.ValueOf(true)
```

Comment:
Perhaps there should be a `flag bool` field on the `spec` struct to indicate whether the field is a boolean flag. `synopsis` could then use this instead of reading `.dest`. The reason I'm nervous about setting `dest := reflect.VaueOf(true)` is that here `dest` won't be assignable (ie calling `dest.Set(...)` will panic), but other bits of the code assume that `spec.dest` is always assignable.